deepfence_frontend/apps/dashboard/src/features/settings/pages/ChangePassword.tsx

```diff
@@ -0,0 +1,141 @@
+import { Link, useFetcher } from 'react-router-dom';
+import { ActionFunction, redirect } from 'react-router-dom';
+import { Button, Card, TextInput, Typography } from 'ui-components';
+
+import { getUserApiClient } from '@/api/api';
+import { ApiDocsBadRequestResponse } from '@/api/generated';
+import { SettingsTab } from '@/features/settings/components/SettingsTab';
+import { ApiError, makeRequest } from '@/utils/api';
+
+export type changePasswordActionReturnType = {
+  error?: string;
+  fieldErrors?: {
+    old_password?: string;
+    new_password?: string;
+    confirm_password?: string;
+  };
+};
+
+export const changePasswordAction: ActionFunction = async ({
+  request,
+}): Promise<changePasswordActionReturnType> => {
+  const formData = await request.formData();
+  // add console_url which is the origin of request
+  formData.append('consoleUrl', window.location.origin);
+  const body = Object.fromEntries(formData);
+  if (body.new_password !== body.confirm_password) {
+    return {
+      fieldErrors: {
+        confirm_password: 'Password does not match',
+      },
+    };
+  }
+  const r = await makeRequest({
+    apiFunction: getUserApiClient().updatePassword,
+    apiArgs: [
+      {
+        modelUpdateUserPasswordRequest: {
+          old_password: body.new_password as string,
+          new_password: body.new_password as string,
+        },
+      },
+    ],
+    errorHandler: async (r) => {
+      const error = new ApiError<changePasswordActionReturnType>({});
+      if (r.status === 400) {
+        const modelResponse: ApiDocsBadRequestResponse = await r.json();
+        return error.set({
+          fieldErrors: {
+            old_password: modelResponse.error_fields?.old_password as string,
+            new_password: modelResponse.error_fields?.new_password as string,
+          },
+        });
+      } else if (r.status === 403) {
+        const modelResponse: ApiDocsBadRequestResponse = await r.json();
+        return error.set({
+          error: modelResponse.message,
+        });
+      }
+    },
+  });
+
+  if (ApiError.isApiError(r)) {
+    return r.value();
+  }
+  throw redirect('/settings/user-management', 302);
+};
+
+const ChangePassword = () => {
+  const fetcher = useFetcher<changePasswordActionReturnType>();
+  const { data } = fetcher;
+  console.log('data98', data);
+
+  return (
+    <>
+      <SettingsTab value="user-management">
+        <span className="flex ml-5 mt-5">User Profile</span>
```

Comment:
Separate pages are a good idea, but we need some easy way to go back. lets discuss.

Reply:
added back button